examples/tv-casting-app/tv-casting-common/src/CastingServer.cpp

```diff
@@ -320,24 +321,81 @@ CastingServer::ContentLauncherLaunchURL(TargetEndpointInfo * endpoint, const cha
 
 void CastingServer::DeviceEventCallback(const DeviceLayer::ChipDeviceEvent * event, intptr_t arg)
 {
+    bool runPostCommissioning           = false;
+    chip::NodeId targetPeerNodeId       = 0;
+    chip::FabricIndex targetFabricIndex = 0;
     if (event->Type == DeviceLayer::DeviceEventType::kBindingsChangedViaCluster)
     {
+        ChipLogProgress(AppServer, "CastingServer::DeviceEventCallback kBindingsChangedViaCluster received");
         if (CastingServer::GetInstance()->GetActiveTargetVideoPlayer()->IsInitialized())
         {
+            ChipLogProgress(AppServer,
+                            "CastingServer::DeviceEventCallback already connected to video player, reading server clusters");
             CastingServer::GetInstance()->ReadServerClustersForNode(
                 CastingServer::GetInstance()->GetActiveTargetVideoPlayer()->GetNodeId());
         }
+        else if (CastingServer::GetInstance()->udcInProgress)
+        {
+            ChipLogProgress(AppServer,
+                            "CastingServer::DeviceEventCallback UDC is in progress while handling kBindingsChangedViaCluster");
+            CastingServer::GetInstance()->udcInProgress = false;
```

Comment:
is this the only place where udcInProgress gets set back to false? Does this code path get hit during a regular commissioning flow?

Reply:
It gets set to false here and at line 377 - essentially anytime we want to mark the end of the user directed commissioning flow. 
While line 377 will execute during a regular commissioning flow where the app just commissioned with a new commissioner, line 341 executes when the app begins user directed commissioning but the commissioner had previously commissioned the app.